scripts/ci-apidiff.sh

```diff
@@ -24,7 +24,7 @@ cd "${REPO_ROOT}" || exit
 echo "*** Running go-apidiff ***"
 APIDIFF=$(APIDIFF_OLD_COMMIT="${PULL_BASE_SHA}" make apidiff 2> /dev/null)
 
-if [[ "${APIDIFF}" == *"sigs.k8s.io/cluster-api-provider-azure/api/"* ]] || [[ "${APIDIFF}" == *"sigs.k8s.io/cluster-api-provider-azure/exp/api/"* ]]; then
+if grep -qE '^sigs\.k8s\.io/cluster-api-provider-azure/(exp/)?api/' <<< "$APIDIFF"; then
```

Comment:
I know this isn't a functional change but can we dig up documentation (inherited from k8s standards is sufficient) that outlines that changing *any* API classified as experimental is permitted? I am uncertain if, say, a v1beta1 API is breakable even if it's experimental + feature flag.

Reply:
I think this is the relevant guidance:
- https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
- https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md

I don't see any mention of what's expected when an API and an accompanying feature have differing alpha/beta/GA status, so I think the API-specific guidance is most relevant here. It looks to me like both alpha and beta APIs can make breaking changes between versions, but for beta APIs we need to automate or document how to upgrade.

Reply:
Thanks, I think we're good here!